### DISCUSSION TITLE
Player is marked as spotted after they actually appeared in the view

### DISCUSSION THREAD

**Discussion by thecatontheflat** [saul/demofile#252] [Questions] [answered]
Hey Saul, thank you for the great piece of work!

I am not sure if this is expected behaviour or a bug, but it seems that the Player is marked as spotted much later then they actually appeared in the actual view of another Player.

Please consider the code below:

```
demoFile.on('tickend', e => {
    demoFile.players.forEach(player => {
      if (!player || player.isFakePlayer) return;
      const enemy = player.allSpotted.find(p => !player.isFriendly(p));

      if (enemy) console.log(demoFile.currentTick));
    });
  });
```

When I watch the replay in the game client (`demo_gototick ...`), in that particular tick these players have been in each other's field of view for around 200 frames.

**Comment by saul** [comment, marked as answer]
I believe the spotted logic is based on what the server is telling the client to show in the radar. I imagine when you're watching the demo you'll see the radar show the enemy at the same tick that you see them being spotted with the `demofile` library.

**Comment by thecatontheflat** [comment]
Ah I see, I was confused about this. Thanks for pointing it out!

Can you think or an efficient way of figuring out whether a player actually sees another player? My mistake was to assume that that’s what “spotted” was for.

I guess I’d have to project a vector based on the eye angles towards the other players hitboxes. I know it’s beyond the scope of this library, but could you maybe point me in the right direction as to where can I get the map spacial data (I believe it’s called collision mesh)? Or is there a smarter way of achieving this?

Thank you again for the amazing library!

**Comment by saul** [comment]
There is no efficient way unfortunately. You'd have to simulate the collision model of the entire world (brushes, doors, props, players, etc) to be able to accurately tell when a player is visible to another player.

If that's what you need then I think you're using the wrong library. It would make more sense to write a server plugin (e.g. using SourceMod), play the demos in-game with the `timedemo` command (to run as fast as possible), and record the information you need that way instead.
